test/unit/lib/plugins/aws/package/compile/events/httpApi.test.js

```diff
@@ -204,6 +204,52 @@ describe('lib/plugins/aws/package/compile/events/httpApi.test.js', () => {
       expect(cfStage.Properties.AccessLogSettings).to.have.property('Format');
     });
 
+    it('should not disable default execute-api endpoint', () => {
+      expect(cfApi.Properties.DisableExecuteApiEndpoint).to.equal(undefined);
+    });
+
+    describe('Disable default execute-api endpoint', () => {
+      before(() =>
+        runServerless({
+          fixture: 'httpApi',
+          configExt: {
+            provider: {
+              httpApi: {
+                disableDefaultEndpoint: true,
+              },
+            },
+          },
+          cliArgs: ['package'],
+        }).then(({ awsNaming, cfTemplate }) => {
+          cfApi = cfTemplate.Resources[awsNaming.getHttpApiLogicalId()];
+        })
+      );
+      it('should disable default execute-api endpoint', () => {
```

Comment:
Let's name it ``should supprt `provider.httpApi.name` `` and move it to `Provider properties` section